tests/RulesetEngineTest.php

```diff
@@ -152,82 +150,85 @@ public function testGetViolationsButNoViolations(array $dependenciesAsArray, arr
             $classNameLayerResolver->getLayersByClassName($classInLayer)->willReturn($layers);
         }
 
-        static::assertCount(
-            $expectedCount,
-            (new RulesetEngine())->getViolations(
-                $dependencyResult,
-                $classNameLayerResolver->reveal(),
-                ConfigurationRuleset::fromArray($rulesetConfiguration)
-            )
+        $configuration = Configuration::fromArray([
+            'layers' => [],
+            'paths' => [],
+            'ruleset' => $rulesetConfiguration,
+        ]);
+
+        $context = (new RulesetEngine())->process(
+            $dependencyResult,
+            $classNameLayerResolver->reveal(),
+            $configuration
         );
+
+        static::assertCount($expectedCount, $context->violations());
     }
 
-    public function provideTestGetSkippedViolations(): array
+    public function provideTestGetSkippedViolations(): iterable
     {
-        return [
```

Comment:
can you please dont change the tests (without a comment here) and just add test cases. it's annoying to review all the testcases.

Reply:
It is necessary to change tests as the behaviour has changed + I had to reformat the data provider method for readability

Reply:
can you do the formatting stuff in a dedicated pr (and direct merge it)?

Reply:
Reverted formatting